modules/core/migrate/src/Plugin/migrate/process/Boolean.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Drupal\farm_migrate\Plugin\migrate\process;
+
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+
+/**
+ * This plugin converts values to a boolean.
+ *
+ * @codingStandardsIgnoreStart
+ *
+ * Example usage:
+ * @code
+ * destination:
+ *   plugin: 'entity:log'
+ * process:
+ *   is_movement:
+ *     plugin: boolean
+ *     source: is_movement
+ * @endcode
+
+ * @codingStandardsIgnoreEnd
+ *
+ * @MigrateProcessPlugin(
+ *   id = "boolean"
```

Comment:
Curious if this ID is too generic. Did a quick search but didn't see anything that is conflicting. This was the closest boolean plugin, interesting that it takes a different approach: https://git.drupalcode.org/project/migration_tools/-/blob/8.x-2.x/src/Plugin/migrate/process/ConvertBoolean.php

I like the logic that is implemented here as a more generic approach but it does make we wonder if having an option to provide a "mapping" would be useful.

Reply:
> Curious if this ID is too generic.

It is indeed generic. I was surprised that this was not already implemented in core. If we want to go to the trouble, we could consider turning it into a core patch and proposing it.

I found the `migration_tools` module and its `convert_boolean` plugin in my research. I felt that PHP's native `filter_var($value, FILTER_VALIDATE_BOOLEAN)` was a more elegant approach.

Sources:

- https://www.php.net/manual/en/filter.filters.validate.php
- https://stackoverflow.com/questions/4775294/parsing-a-string-into-a-boolean-value-in-php

I'm not too worried about potential conflicts with future core/contrib plugin IDs. If that happens, then we can cross that bridge... but I would assume that it won't be very difficult to transition.

> I like the logic that is implemented here as a more generic approach but it does make we wonder if having an option to provide a "mapping" would be useful.

I suppose if there's a need to explicitly restrict the available options, then a mapping would be useful. In that case, I think a developer can use the existing `static_map` process plugin. So I don't think we need to provide that option, personally. But if it comes up in the future as a valid need, then I suppose we could append configuration options to this `boolean` plugin that allow an optional mapping to be specified.